tests/returntypes/jax/test_jax_jit_qnode_new.py

```diff
@@ -70,7 +70,7 @@ def circuit(a):
         assert circuit.interface == interface
 
         # the tape is able to deduce trainable parameters
-        assert circuit.qtape.trainable_params == [0]
+        assert circuit.qtape.trainable_params == []
```

Comment:
What was the previous behaviour doing here? Why was the gradient empty if we explicitly set a trainable parameter? Was it ignored because of JITting?

Reply:
Jitting would make the first parameter a tracer and therefore we would end up identify it as trainable. But the gradient is not empty.

```
import pennylane as qml
import pennylane.numpy as np

import jax
from jax.config import config

remember = config.read("jax_enable_x64")
config.update("jax_enable_x64", True)

qml.enable_return()
dev = qml.device("default.qubit", wires=2)

@qml.qnode(dev, interface="jax", diff_method="parameter-shift")
def circuit(a):
    qml.RY(a, wires=0)
    qml.RX(0.2, wires=0)
    return qml.expval(qml.PauliZ(0))
a = np.array(0.1, requires_grad=True)

print(jax.jit(circuit)(a))
print(circuit.tape.trainable_params)
```
```
0.9751703272018156
[0]
```